Sources/PointFree/Blog/BlogMiddleware.swift

```diff
@@ -0,0 +1,36 @@
+import ApplicativeRouterHttpPipelineSupport
+import Either
+import Foundation
+import HttpPipeline
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+let blogMiddleware =
+  basicAuth(
+    user: AppEnvironment.current.envVars.basicAuth.username,
+    password: AppEnvironment.current.envVars.basicAuth.password
+    )
+    <| _blogMiddleware
+
+private func _blogMiddleware(
+  conn: Conn<StatusLineOpen, Tuple4<Database.User?, SubscriberState, Route, Route.Blog>>
+  ) -> IO<Conn<ResponseEnded, Data>> {
+
+  let (user, subscriberState, route, subRoute) = lower(conn.data)
+
+  switch subRoute {
```

Comment:
it's kinda nice to move all the blog sub routes into its own file instead of making `SiteMiddleware` bigger and bigger. I did this just so that I could more easily add basic auth, but I think we could more often break out large sub routes into their own middlewares.